### PR TITLE
Remove test helpers

### LINKED ISSUE #28: Move startMirage initializer into helper
import { startMirage } from '../../initializers/ember-cli-mirage';

This is used in a lot of our tests. Probably should put this in a test helper.

### LINKED ISSUE #319: Remove visitWithoutWait & waitForElement helpers after upgrading to the latest ember-test-helpers

### PR DESCRIPTION
**Pull request by kevinhinterlong**
Resolves #28 
Resolves #319 

## Description
This makes most of our tests follow the same flow and gets rid of unneeded helpers
Not great that we use `this.server`, but better than accidentally forgetting to shutdown

## Proposed Changes
Remove unneeded test helpers, switch most tests over to setupMirage

## License
I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/core/tests/unit/initializers/inject-c3-enhancements-test.js [hunk 1]
```diff
@@ -3,22 +3,22 @@ import { run } from '@ember/runloop';
 import { initialize } from 'dummy/initializers/inject-c3-enhancements';
 import { module, test } from 'qunit';
 import c3 from 'c3';
-import destroyApp from '../../helpers/destroy-app';
 
+let App;
 module('Unit | Initializer | inject c3 enhancements', function(hooks) {
   hooks.beforeEach(function() {
     run(() => {
```
Comment (jkusa) [inline packages/core/tests/unit/initializers/inject-c3-enhancements-test.js]: Looks like the blueprint takes this approach:
```js
  hooks.beforeEach(function() {
    this.TestApplication = Application.extend();
    this.TestApplication.initializer({
      name: 'initializer under test',
      initialize
    });

    this.application = this.TestApplication.create({ autoboot: false });
  });

  hooks.afterEach(function() {
    run(this.application, 'destroy');
  });
```

File: packages/core/tests/unit/models/bard-request/fragments/filter-test.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ module('Unit | Model Fragment | BardRequest - Filter', function(hooks) {
   setupTest(hooks);
 
   hooks.beforeEach(async function() {
-    Server = setupMock();
+    Server = startMirage();
```
Comment (jkusa) [inline packages/core/tests/unit/models/bard-request/fragments/filter-test.js]: Can we use `setupMirage(hooks)` here?

Comment (jkusa) [reply packages/core/tests/unit/models/bard-request/fragments/filter-test.js]: And use `this.server` instead of `let Server`

Comment (kevinhinterlong) [reply packages/core/tests/unit/models/bard-request/fragments/filter-test.js]: Yea we could switch everything to using `setupMirage(hooks)`, I just wasn't a fan of the `this.server` being magically set from [here](https://github.com/samselikoff/ember-cli-mirage/blob/c77f1997701fbc476cade1717197d48a0bcd70b9/addon-test-support/setup-mirage.js#L13)

File: packages/core/tests/unit/models/mock-test.js [hunk 3]
```diff
@@ -3,24 +3,25 @@ import { run } from '@ember/runloop';
...
+import { startMirage } from 'dummy/initializers/ember-cli-mirage';
+
+let Server;
 
 module('Unit | Adapter | base json adapter', function(hooks) {
   setupTest(hooks);
 
   hooks.beforeEach(function() {
-    setupMock();
+    Server = startMirage();
```
Comment (jkusa) [inline packages/core/tests/unit/models/mock-test.js]: hook?

File: packages/core/tests/unit/models/report-test.js [hunk 4]
```diff
@@ -78,14 +78,14 @@ module('Unit | Model | report', function(hooks) {
   setupTest(hooks);
 
   hooks.beforeEach(async function() {
-    server = setupMock();
+    Server = startMirage();
```
Comment (jkusa) [inline packages/core/tests/unit/models/report-test.js]: hook

File: packages/core/tests/unit/services/user-test.js [hunk 5]
```diff
@@ -1,20 +1,20 @@
...
-let Store, NaviUser, server;
+let Store, NaviUser, Server;
 
 module('Unit | Service | user', function(hooks) {
   setupTest(hooks);
 
   hooks.beforeEach(function() {
     NaviUser = config.navi.user;
-    server = setupMock();
+    Server = startMirage();
```
Comment (jkusa) [inline packages/core/tests/unit/services/user-test.js]: hook

File: packages/reports/tests/integration/components/dimension-bulk-import-test.js [hunk 6]
```diff
@@ -48,8 +48,8 @@ module('Integration | Component | Dimension Bulk Import', function(hooks) {
   setupRenderingTest(hooks);
 
   hooks.beforeEach(function() {
-    server = setupMock();
-    server.get(
+    Server = startMirage();
```
Comment (jkusa) [inline packages/reports/tests/integration/components/dimension-bulk-import-test.js]: hook

File: packages/reports/tests/integration/components/filter-values/dimension-select-test.js [hunk 7]
```diff
@@ -21,19 +21,19 @@ const MockFilter = {
...
-let server;
+let Server;
 
 module('Integration | Component | filter values/dimension select', function(hooks) {
   setupRenderingTest(hooks);
 
   hooks.beforeEach(function() {
     this.filter = MockFilter;
-    server = setupMock();
+    Server = startMirage();
```
Comment (jkusa) [inline packages/reports/tests/integration/components/filter-values/dimension-select-test.js]: hook

File: packages/reports/tests/unit/consumers/item-test.js [hunk 8]
```diff
@@ -11,15 +11,15 @@ module('Unit | Consumer | item', function(hooks) {
   setupTest(hooks);
 
   hooks.beforeEach(function() {
-    Server = setupMock();
+    Server = startMirage();
```
Comment (jkusa) [inline packages/reports/tests/unit/consumers/item-test.js]: hook

File: packages/core/tests/unit/models/bard-request/fragments/filter-test.js (hunk 2 above)
Comment (jkusa) [reply packages/core/tests/unit/models/bard-request/fragments/filter-test.js]: > I just wasn't a fan of the this.server being magically set from here

Yeah, I'm with you, but I think automatic `this.server.shutdown()` via hooks is worth the trade-off.

File: packages/core/tests/unit/initializers/inject-c3-enhancements-test.js [hunk 9]
```diff
@@ -1,24 +1,29 @@
...
 import { run } from '@ember/runloop';
-import { initialize } from 'dummy/initializers/inject-c3-enhancements';
+import c3Enhancements from 'dummy/initializers/inject-c3-enhancements';
 import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
 import c3 from 'c3';
-import destroyApp from '../../helpers/destroy-app';
 
 module('Unit | Initializer | inject c3 enhancements', function(hooks) {
+  setupTest(hooks);
```
Comment (balajimanoharan) [inline packages/core/tests/unit/initializers/inject-c3-enhancements-test.js]: Should we make this an acceptance test instead?

**Pull request merged**